### DISCUSSION TITLE
Early support for browsers that the output will be expected to work on

### DISCUSSION THREAD

**Discussion by Erudition** [webstudio-is/webstudio-community#89] [Ideas]
While there are FLO versions of the proprietary Google Chrome (e.g. `ungoogled-chromium`), Firefox is the browser of choice for many privacy and FLO advocates and I expect the sites I build to be able to work consistently across browsers - in fact the design tool is a great place to limit features to those supported cross-browser and cross-platform.

As it currently stands with Webstudio only supporting Blink-based browsers, I naturally get the (hopefully false) impression that the site I'm designing can only be relied on to look/work as expected in those same browsers.

Perhaps it's just the app's UI that is Chrome-specific and the tool's output is cross-browser, but when cross-browser support is not targeted early on it's often the case that browser-specific features creep in that need to be substantially refactored when the decision to support more browsers is finally made, so the time saved upfront is often not worth it...

Anyway, if other browsers are targeted with Webstudio's output, I suggest mentioning that in the tool's not-supported error message, such as:

> Sites produced with Webstudio have cross-platform support, but the Webstudio app itself currently runs only on Blink-based browsers like Google Chrome and Microsoft Edge. We plan to support more browsers in the near future.

**Comment by kof** [comment]
Yeah, its a false impression, builder != "your site", its a good idea to mention it there, I saw several people jumping to the wrong conclusions too!

**Reply by kof** [reply]
Hope this fixes it https://github.com/webstudio-is/webstudio/pull/2686

**Reply by Erudition** [reply]
Looks good! Mind if I ask what prevents Webstudio from running on Firefox? It seems to load fine in the background of that message

**Reply by kof** [reply]
various bugs you don't see at the first glance, we need to organize all issues with labels

**Reply by ghost** [reply]
It's a shame Firefox is not supported. Just wanted to start using this software as a complete replacement for Webflow. Ah well, I'll check in again in a few months' time!

**Reply by kof** [reply]
we started aggregating all bugs related to various browsers in his list https://github.com/webstudio-is/webstudio/issues/1818

**Reply by kof** [reply]
We recently even discovered a bug in chromium 118 which is still being used in our electron app https://github.com/webstudio-is/webstudio/issues/3331

Now if we can't even support all widely used chromium versions, imagine how hard its gonna be to support dieing browsers like firefox

We started aggregating all the bugs

https://github.com/webstudio-is/webstudio/issues/1818


The best we can do is enable using the browser despite known bugs after user acknowledges this information. I don't think we are going to be supporting many browsers inside builder any time soon.

Builder is for the best building experience, it needs to be able to use latest and greatest features browser has to offer.

**Reply by Erudition** [reply]
Interesting take.

I don't see Firefox as "dying", it's been floating around the same 5-10% market share for the last 5 years, higher than nearly all other browsers except the two proprietary juggernauts. I'm a web developer, and I use free/libre/open software (hence Webstudio, hence Firefox) when possible. Both of these traits are likely overrepresented in your community, so the market share doesn't even tell the whole story here.

I currently rely on Elementor, which works just as well and with a similar number of bugs in Firefox as with Chrome. Both browsers have support some of the "latest and greatest" standards, with FF often being more standards-compliant. I guess the cutting edge features you are referring to needing are not standards though.

**Reply by kof** [reply]
Chromium is also open source and is used by a number of other browsers.

**Reply by kof** [reply]
Firefox is often lagging behind specs, because underfunded, but this is also a question if we have time to invest into dealing with those situations instead of building features in one browser that we support really well.

**Reply by ghost** [reply]
Looking back at the attitude of the Webstudio devs not supporting Firefox, it seems like I have made the better decision to go with alternative solutions.

**Reply by kof** [reply]
There is no attitude, there is limited resources, small team. We need to make tradeoffs right now in order to advance the product for majority of users.

That being said we just added ability to test the site in preview mode in all browsers. So only building itself is still limited to chromium based browsers.

**Reply by Aasikki** [reply]
> We recently even discovered a bug in chromium 118 which is still being used in our electron app [webstudio-is/webstudio#3331](https://github.com/webstudio-is/webstudio/issues/3331)
> 
> Now if we can't even support all widely used chromium versions, imagine how hard its gonna be to support dieing browsers like firefox
> 
> We started aggregating all the bugs
> 
> [webstudio-is/webstudio#1818](https://github.com/webstudio-is/webstudio/issues/1818)
> 
> The best we can do is enable using the browser despite known bugs after user acknowledges this information. I don't think we are going to be supporting many browsers inside builder any time soon.
> 
> Builder is for the best building experience, it needs to be able to use latest and greatest features browser has to offer.

Dying browser lmao. Now I see why you only support Chrome. It's entirely your attitude. Your project looked cool but I think I'll have to pass. Was even ready to use chrome but nah, not with that kind of take. Can't trust you won't have that attitude towards other things. Sorry.
